### PR TITLE
Bump cheatsheet functionality and fill a few gaps

### PR DESCRIPTION
**Pull request by mschwager**
We're pretty close to parity with our officially support languages except for a few gaps:

- JS import aliasing, this is enough of a painful mess with all the different `import`/`require` features that I'll tackle this separately. Tracking here: #285
- "Import" functionality in C - tracking here: #1095 
- Ruby still has quite a few gaps, I'll tackle those in a separate change

### CODE REVIEW

**Comment by mschwager** [comment]
See also #1317, #1318, and #1319 tracking separately.

File: semgrep-core/tests/generate_test_matrix.py [hunk 1]
```diff
@@ -40,6 +40,15 @@
     "exprstmt": "Expression and Statement",
 }
 
+LANGUAGE_EXCEPTIONS = {
+    "java": [
+        "naming_import",
```
Comment (ievans) [inline semgrep-core/tests/generate_test_matrix.py:45]: Is there an issue tracking this?

File: semgrep-core/tests/generate_test_matrix.py [hunk 2]
```diff
@@ -89,7 +98,14 @@ def generate_cheatsheet(root_dir: str):
...
                 entry = (read_if_exists(sgrep_path), sgrep_path,  read_if_exists(code_path), code_path)
                 print((lang, entry))
-                output[lang][VERBOSE_FEATURE_NAME.get(category, category)][VERBOSE_SUBCATEGORY_NAME.get(subcategory, subcategory)].append(entry)
+                feature_name = VERBOSE_FEATURE_NAME.get(category, category)
+                subcategory_name = VERBOSE_SUBCATEGORY_NAME.get(subcategory, subcategory)
+                language_exception = (
+                    feature_name in LANGUAGE_EXCEPTIONS.get(lang, [])
+                    or subcategory in LANGUAGE_EXCEPTIONS.get(lang, [])
+                )
+                if not language_exception:
```
Comment (ievans) [inline semgrep-core/tests/generate_test_matrix.py:107]: is it really worth having language exception? I'd prefer to be explicit: "we don't have test cases for this, it is probably not supported"

**Review by ievans — APPROVED** [review]

File: semgrep-core/tests/generate_test_matrix.py (hunk 1 above)
Comment (mschwager) [reply semgrep-core/tests/generate_test_matrix.py:45]: From what I can tell, Java doesn't support import aliasing. To address the comment below as well, this list _is_ the explicit declaration that a language doesn't support something. A missing test tells me "someone forgot to test this, or maybe it's not supported" whereas this list tells me "this language doesn't support this feature." Or, at least, that's what I was hoping. Is there a better way we can state that?

**Pull request merged**